cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/AbstractStreamFileConsumerFactory.java

```diff
@@ -163,14 +165,23 @@ public void dropAll(QueueName streamName, String namespace, Iterable<Long> group
       streamAdmin.configureGroups(streamName, groupInfo);
 
       if (oldStreamAdmin instanceof QueueAdmin && !oldStreamAdmin.exists(streamName.toURI().toString())) {
-        // A bit hacky to assume namespace is formed by appId.flowId. See AbstractDataFabricFacade
-        // String namespace = String.format("%s.%s", programId.getApplicationId(), programId.getId());
-
-        int idx = namespace.indexOf('.');
-        String appId = namespace.substring(0, idx);
-        String flowId = namespace.substring(idx + 1);
-
-        ((QueueAdmin) oldStreamAdmin).dropAllForFlow(appId, flowId);
+        // A bit hacky to assume namespace is formed by namespaceId.appId.flowId. See AbstractDataFabricFacade
+        // String namespace = String.format("%s.%s.%s",
+        //                                  programId.getNamespaceId(),
+        //                                  programId.getApplicationId(),
+        //                                  programId.getId());
+
+        String invalidNamespaceError = String.format("Namespace string %s must be of the form <namespace>.<app>.<flow>",
+                                                     namespace);
+        Iterator<String> namespaceParts = Splitter.on('.').split(namespace).iterator();
+        Preconditions.checkArgument(namespaceParts.hasNext(), invalidNamespaceError);
+        String namespaceId = namespaceParts.next();
+        Preconditions.checkArgument(namespaceParts.hasNext(), invalidNamespaceError);
+        String appId = namespaceParts.next();
+        Preconditions.checkArgument(namespaceParts.hasNext(), invalidNamespaceError);
+        String flowId = namespaceParts.next();
+
+        ((QueueAdmin) oldStreamAdmin).dropAllForFlow(namespaceId, appId, flowId);
```

Comment:
Carried over from old code. Needed because ``oldStreamAdmin`` is declared to be of type ``StreamAdmin``, and is cast to ``QueueAdmin`` for doing the *queue* related work here.